src/scrape_up/kooapp/users.py

```diff
@@ -0,0 +1,130 @@
+            return userdata
+        except Exception as e:
+            raise Exception(f"An error occured while parsing the data: {e}")
+
+    def get_name(self) -> str:
+        """
+        Fetch the name of the user.
+        """
+        userdata = self.__parse_profile_data()
+        name = userdata.get('name')
+        return {
+            'data': name or None,
+            'message': f"Name of the user {self.username} is {name}" if name else f"No name found for the user {self.username}"
+        }
+
+    def get_bio(self) -> str:
+        """
+        Fetch the bio dsecription of the user.
+        """
+        userdata = self.__parse_profile_data()
+        bio = userdata.get('description')
+        return {
+            'data': bio or None,
+            'message': f"Bio found for the user {self.username}" if bio else f"No bio found for the user {self.username}"
+        }
+    
+    def get_avatar_url(self) -> str:
+        """
+        Fetch the avatar url of the user.
+        """
+        userdata = self.__parse_profile_data()
+        avatar = userdata.get("profileImage")
+        return {
+            'data': avatar or None,
+            'message': f"Avatar found for the user {self.username}" if avatar else f"Avatar not found for the user {self.username}"
+        }
+    
+    def followers(self) -> int:
+        """
+        Fetch the number of followers of the Koo user.
+        """
+        userdata = self.__parse_profile_data()
+        followers = userdata.get("followerCount")
+        return {
+            'data': followers,
+            'message': f"There are {followers} follower(s) of the user {self.username}"
+        }
+    
+    def following(self) -> int:
+        """
+        Fetch the number of following of the Koo user.
+        """
+        userdata = self.__parse_profile_data()
+        following = userdata.get("followingCount")
+        return {
+            'data': following,
+            'message': f"There are {following} user(s) followed by {self.username}"
+        }
+    
+    def get_social_profiles(self) -> dict[str, str]:
+        """
+        Fetch all the listed social media profiles of user.
+        """
+        userdata = self.__parse_profile_data()
+        profiles = {s_media: handle for s_media, handle in userdata.get('socialProfile').items() if handle }
+        return {
+            'data': profiles or None,
+            'message': f"Found {len(profiles)} social profiles for the user {self.username}" if profiles else f"No social profiles found for the user {self.username}"
+        }
+
+    def get_profession(self) -> str:
+        """
+        Fetch the profession of the user.
+        """
+        userdata = self.__parse_profile_data()
+        profession = userdata.get('title')
+        return {
+            'data': profession or None,
+            'message': f"Profession found for the user {self.username}" if profession else f"No profession found for the user {self.username}"
+        }
```

Comment:
Try to implement a try-catch. It is a good practice and also prevents unexpected errors.
An ideal method looks like ...
![image](https://github.com/Clueless-Community/scrape-up/assets/93156825/f9484188-ab70-4aa1-919f-db43ef214178)

Reply:
There is no use of using try catch block.. errors are already being handled and since i am using dictionary... `.get` method will return None instead of raising any KeyError.

Reply:
Okay then !!